### PR TITLE
recipes-zarhus: add alsa to separate packagegroups

### PR DESCRIPTION
**Pull request by dancios**
This patch adds packagegroups:
* zarhus-alsa
* zarhus-alsa-tools

### CODE REVIEW

**Comment by DaniilKl** [comment]
@dancios, you should mark a PR as a draft if it is not ready for review.

File: recipes-zarhus/packagegroups/packagegroup-zarhus.bb [hunk 1]
```diff
@@ -24,3 +26,24 @@ RDEPENDS:${PN}-dbg = " \
...
+    dialog \
+    ncurses \
+"
+
+RDEPENDS:${PN}-alsa-tools = " \
+    alsa-utils-speakertest \
+    mpg123 \
+    gettext \
+    dialog \
+    ncurses \
```
Comment (m-iwanicki) [inline recipes-zarhus/packagegroups/packagegroup-zarhus.bb]: You can

```suggestion
    ${PN}-alsa \
```

if alsa-tools depends on alsa

File: recipes-zarhus/packagegroups/packagegroup-zarhus.bb [hunk 2]
```diff
@@ -24,3 +26,24 @@ RDEPENDS:${PN}-dbg = " \
     libgpiod-tools \
     devmem2 \
 "
+RDEPENDS:${PN}-alsa = " \
+    alsa-utils \
```
Comment (m-iwanicki) [inline recipes-zarhus/packagegroups/packagegroup-zarhus.bb]: Might be useful?

```suggestion
RDEPENDS:${PN}-alsa = " \
    packagegroup-base-alsa \
    alsa-utils \
```

Might need `alsa` in MACHINE_FEATURES if it isn't there. Or just copy https://git.yoctoproject.org/poky/plain/meta/recipes-core/packagegroups/packagegroup-base.bb and add

```
${@bb.utils.contains("MACHINE_FEATURES", "alsa", "packagegroup-zarhus-alsa", "", d)} \
${@bb.utils.contains("MACHINE_FEATURES", "alsa", "packagegroup-zarhus-alsa-tools", "", d)} \ 
```

to PACKAGES

File: recipes-zarhus/packagegroups/packagegroup-zarhus.bb (hunk 1 above)
Comment (m-iwanicki) [reply recipes-zarhus/packagegroups/packagegroup-zarhus.bb]: Or maybe the other way include alsa-tools in alsa (though I think alsa-tools depends on alsa seems more correct)

File: recipes-zarhus/packagegroups/packagegroup-zarhus.bb [hunk 3]
```diff
@@ -24,3 +26,24 @@ RDEPENDS:${PN}-dbg = " \
...
+    alsa-plugins \
+    alsa-lib \
+    alsa-tools \
+    alsa-state \
+    libasound \
+    alsa-utils-speakertest \
+    mpg123 \
+    gettext \
+    dialog \
+    ncurses \
```
Comment (m-iwanicki) [inline recipes-zarhus/packagegroups/packagegroup-zarhus.bb]: Is this needed? Usually recipes that need this to run should have this in RDEPENDS in their own recipe.

**Review by m-iwanicki — CHANGES_REQUESTED** [review]

**Comment by macpijan** [comment]
Thanks for comments @m-iwanicki , I have done some very basic cleanup here and moved the rest into issue: https://github.com/zarhus/zarhus-issues/issues/28

**Pull request merged**